app/tls.go

```diff
@@ -108,7 +108,6 @@ func SimpleDialTLSConfig(cert tls.Certificate, pool *x509.CertPool) *tls.Config
 	if len(x509cert.DNSNames) == 0 {
 		panic("certificate has no DNS extension")
 	}
-	config.ServerName = x509cert.DNSNames[0]
```

Comment:
I think this one should be kept, since the goal of the "simple" TLS configuration is to support sharing the same certificate across nodes. If a user wants per-node certificates it should not use these helpers and build his own `*tls.Config` object from scratch.